### PR TITLE
Use OpenAPI to Document Endpoints

### PR DESCRIPTION
**Pull request by trobanga**

### CODE REVIEW

File: clinical-domain-agent/pom.xml [hunk 1]
```diff
@@ -32,6 +32,24 @@
...
 
+    <dependency>
+      <groupId>io.swagger.core.v3</groupId>
+      <artifactId>swagger-core</artifactId>
+      <version>2.2.26</version>
+    </dependency>
+
+    <dependency>
+      <groupId>jakarta.xml.bind</groupId>
+      <artifactId>jakarta.xml.bind-api</artifactId>
```
Comment (knoppiks) [inline clinical-domain-agent/pom.xml:43]: Spring boot dependencies already specifies `jakarta.xml.bind-api`s version, see [maven central](https://mvnrepository.com/artifact/org.springframework.boot/spring-boot-dependencies/3.4.0). Therefor we should be able to omit the explicit version here.

**Review by knoppiks — CHANGES_REQUESTED** [review]

File: clinical-domain-agent/pom.xml (hunk 1 above)
Comment (trobanga) [reply clinical-domain-agent/pom.xml:43]: No, we need this exact version. Without this swagger-core throws an exception. Looks like a bug in swagger-core and/or springdoc-openapi-starter-webflux-ui to me.

File: trust-center-agent/src/main/java/care/smith/fts/tca/rest/DeIdentificationController.java [hunk 2]
```diff
@@ -53,6 +70,17 @@ private static Mono<ResponseEntity<TransportMappingResponse>> handleGenerateErro
...
+  @Operation(
+      summary = "Get the research mapping",
+      requestBody =
+          @io.swagger.v3.oas.annotations.parameters.RequestBody(
+              content = @Content(schema = @Schema(implementation = String.class))),
+      responses = {
+        @ApiResponse(
+            responseCode = "200",
+            content = @Content(schema = @Schema(implementation = ResearchMappingResponse.class))),
+        @ApiResponse(responseCode = "500", description = "Internal server error")
```
Comment (knoppiks) [inline trust-center-agent/src/main/java/care/smith/fts/tca/rest/DeIdentificationController.java:82]: Should we really mention 500?

File: research-domain-agent/src/main/java/care/smith/fts/rda/rest/TransferProcessController.java [hunk 3]
```diff
@@ -144,12 +192,45 @@ private BodyBuilder responseForStatus(Status s) {
...
+                          name = "Example configuration",
+                          value =
+                              "{\"cohortSelector\":{\"trustCenterAgent\":{\"server\":{\"baseUrl\":\"http://tc-agent:8080\"},\"domain\":\"MII\",\"patientIdentifierSystem\":\"https://ths-greifswald.de/fhir/gics/identifiers/Pseudonym\",\"policySystem\":\"https://ths-greifswald.de/fhir/CodeSystem/gics/Policy\",\"policies\":[\"IDAT_erheben\",\"IDAT_speichern_verarbeiten\",\"MDAT_erheben\",\"MDAT_speichern_verarbeiten\"]}},\"dataSelector\":{\"everything\":{\"fhirServer\":{\"baseUrl\":\"http://cd-hds:8080/fhir\"},\"resolve\":{\"patientIdentifierSystem\":\"http://fts.smith.care\"}}},\"deidentificator\":{\"deidentifhir\":{\"trustCenterAgent\":{\"server\":{\"baseUrl\":\"http://tc-agent:8080\"},\"domains\":{\"pseudonym\":\"MII\",\"salt\":\"MII\",\"dateShift\":\"MII\"}},\"maxDateShift\":\"P14D\",\"deidentifhirConfig\":\"/app/projects/example/deidentifhir/CDtoTransport.profile\",\"scraperConfig\":\"/app/projects/example/deidentifhir/IDScraper.profile\"}},\"bundleSender\":{\"researchDomainAgent\":{\"server\":{\"baseUrl\":\"http://rd-agent:8080\"},\"project\":\"example\"}}}")
```
Comment (knoppiks) [inline research-domain-agent/src/main/java/care/smith/fts/rda/rest/TransferProcessController.java:230]: Would a multi-mine string increase readability?

File: .github/scripts/openapi-specs.sh [hunk 4]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+set -euo pipefail
+
+if ! base_url="http://$(docker compose ${2:-} port "${1}" 8080)"; then
```
Comment (knoppiks) [inline .github/scripts/openapi-specs.sh:4]: Please use var arg expansion here, if possible:
```suggestion
if ! base_url="http://$(docker compose ${@:2} port "${1}" 8080)"; then
```

File: .github/workflows/build.yml [hunk 5]
```diff
@@ -351,9 +351,7 @@ jobs:
...
 
     - name: Trigger Pages Build
-      if: github.ref_type == 'tag'
-      # We have to build the pages after the release,
-      # so that the Pages workflow can find the latest release tag
       env:
         GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-      run: gh workflow run Pages --ref main --repo ${{ github.repository }}
+      # ref could be main
+      run: gh workflow run Pages --ref ${{ github.ref }} --repo ${{ github.repository }} -f build_run_id=${{ github.run_id }}
```
Comment (knoppiks) [inline .github/workflows/build.yml:357]: Do we still need to pass the run_id to the pages workflow?

**Review by knoppiks — CHANGES_REQUESTED** [review]

File: pom.xml [hunk 6]
```diff
@@ -71,6 +81,18 @@
         <artifactId>mockserver-junit-jupiter</artifactId>
         <version>5.15.0</version>
       </dependency>
+
+      <!-- Mitigate conflict with transitive org.mock-server/swagger dependency -->
```
Comment (knoppiks) [inline pom.xml:85]: We no longer use mock-server, do we still need this mitigation?

**Review by knoppiks — DISMISSED** [review]

**Pull request merged**